tutorials/graphql/intro-graphql/tutorial-site/content/graphql-mutations.md

```diff
@@ -1,66 +1,95 @@
+title: GraphQL Mutations - Writing data
 metaTitle: "GraphQL Mutations to insert data | GraphQL Tutorial"
 metaDescription: "Try out GraphQL Mutation using GraphiQL. A GraphQL mutation example with dynamic arguments and variables to insert data"
 ---
 
 import {Link} from "gatsby";
 
-These are the concepts you should know before you attack mutations (haha):
-- <Link to="/graphql-queries/#graphiql">Using GraphiQL</Link>
-- <Link to="/graphql-queries/#graphqlvariables:passingargumentstoyourqueriesdynamically">Using query variables</Link>
+Before learning about GraphQL Mutations, you should be familiar with the following concepts:
+- <Link to="/graphql-queries/#graphiql">GraphiQL</Link>
+- <Link to="/graphql-queries/#graphqlvariables:passingargumentstoyourqueriesdynamically">Query Variables</Link>
 
-Now, let's get started with seeing how we can use GraphQL to "write" data.
-GraphQL mutations are types of GraphQL queries that may result in the state
-of your backend "mutating" or changing, just like typical `'POST'`,
-`'PUT'`, `'PATCH'`, `'DELETE'` APIs.
+Like GraphQL Queries, Mutations use variables too, so it's important to be familiar with them.
 
-## Basic mutations {#basic-mutations}
-Since we're using Hasura for our GraphQL API, we get mutations for
-inserts, updates or deletes that we can use in our app.
+## What is a GraphQL Mutation? {#what-is-a-graphql-mutation}
 
-Let's try these mutations out in the context of a todo app to see
-what mutations look like. Mutations that you get from another GraphQL
-service, say if your API team has built their own,  might be different.
+Up to this point, you only learned how to fetch data in GraphQL. But the real-world applications are complex and you also need to insert, update or delete data. The question is - how do you do it?
+
+In GraphQL, you insert, update or delete data with **mutations**. A Mutation is a GraphQL Operation that allows you to insert new data or modify the existing data on the server-side. You can think of GraphQL Mutations as the equivalent of `POST`, `PUT`, `PATCH` and `DELETE` requests in REST.
+
+To understand better, let’s look at a GraphQL Mutation example:
+
+```graphql
+mutation {
+  insert_todos(objects: [{ title: "Learn GraphQL" }]) {
+    returning {
+      id
+      created_at
+    }
+  }
+}
+```
+
+The above mutation inserts a new todo note into the database. If successful, it returns the `id` and the `creation date` of the newly inserted todo.
+
+What can we observe by looking at the above mutation?
+1. It uses the keyword `mutation` rather than `query`.
+2. The mutation takes an input. We provide the title of the new todo - `Learn GraphQL`.
+3. The mutation returns data. It returns the specified fields for the newly inserted todo.
+
+You will see more GraphQL Mutations later and you will also have the chance to try them in the GraphiQL IDE.
+
+## Types of GraphQL Mutations {#types-of-graphql-mutations}
+
+There are three types of mutations in GraphQL:
```

Comment:
We can rephrase this to `Mutations in GraphQL can be categorised under the following types.

The reason is, GraphQL spec doesn't have `types` of mutations in particular. It is just our nomenclature.